packages/client/src/object/convertWireToOsdkObjects.ts

```diff
@@ -112,6 +116,80 @@ export async function convertWireToOsdkObjects(
   return ret;
 }
 
+export async function convertWireToOsdkObjects2(
+  client: MinimalClient,
+  objects: OntologyObjectV2[],
+  interfaceApiName: string | undefined,
+  forceRemoveRid: boolean = false,
+  selectedProps?: ReadonlyArray<string>,
+  strictNonNull: NullabilityAdherence = false,
+  interfaceToObjectTypeMappings: Record<
+    InterfaceTypeApiName,
+    InterfaceToObjectTypeMappings
+  > = {},
+): Promise<Osdk.Instance<ObjectOrInterfaceDefinition>[]> {
+  client.logger?.debug(`START convertWireToOsdkObjects2()`);
+
+  fixObjectPropertiesInPlace(objects, forceRemoveRid);
+
+  const ret = [];
+  for (const rawObj of objects) {
+    const objectDef = await client.ontologyProvider.getObjectDefinition(
+      rawObj.$apiName,
+    );
+    invariant(objectDef, `Missing definition for '${rawObj.$apiName}'`);
+
+    const interfaceToObjMapping = interfaceApiName
+      ? interfaceToObjectTypeMappings[interfaceApiName as InterfaceTypeApiName][
+        rawObj.$apiName
+      ]
+      : undefined;
+
+    const ifaceSelected = interfaceApiName && interfaceToObjMapping
+      ? (selectedProps
+        ? Object.keys(interfaceToObjMapping).filter(
+          val => {
+            selectedProps?.includes(interfaceToObjMapping[val]);
+          },
+        )
+        : [
+          ...Object.values(interfaceToObjMapping),
+          objectDef.primaryKeyApiName,
```

Comment:
All the results from this API come in as the object props, but scoped down to whats on the interface. As a result, if the primary key isn't part of the interface, we don't get the api name of the primary key from the mapping in the result. We also can't just use the keys of the `objectDef` because that would include props not implemented in the interface